### PR TITLE
Seg

### PR DESCRIPTION
**Pull request by chunhuizhang**

### CODE REVIEW

File: example/fcn-xs/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+FCN-xs EXAMPLES
+---------------
+This folder contains the examples of image segmentation in MXNet.
+
+## Sample results
+![fcn-xs pasval_voc result](https://github.com/tornadomeet/mxnet/blob/seg/example/fcn-xs/fcn-xs_pascal.jpg)
```
Comment (vchuravy) [inline example/fcn-xs/README.md]: This url should point to the image included in this PR

File: dmlc-core [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit a9b3320d2c6b29506139784b877142c9ee78caaf
+Subproject commit 4b951c0378386b7f4d9eae72be2ecd3b9c816afe
```
Comment (vchuravy) [inline dmlc-core]: 4b951c is older than a9b332. Accidential commit?

File: src/operator/softmax_output-inl.h [hunk 3]
```diff
@@ -88,7 +92,7 @@ class SoftmaxOutputOp : public Operator {
       Tensor<xpu, 2> label = in_data[softmaxout_enum::kLabel].FlatTo2D<xpu, real_t>(s);
       Tensor<xpu, 3> out = out_data[softmaxout_enum::kOut].get_with_shape<xpu, 3, real_t>(s3, s);
       Tensor<xpu, 3> grad = in_grad[softmaxout_enum::kData].get_with_shape<xpu, 3, real_t>(s3, s);
-      SoftmaxGrad(grad, out, label);
+      SoftmaxGrad(grad, out, label, static_cast<real_t>(param_.ignore_label));
```
Comment (vchuravy) [inline src/operator/softmax_output-inl.h]: Maybe I am missing something, but shouldn't this be accompanied by a change to mshadow?

**Comment by vchuravy** [comment]
Thanks :) I added some comments, but somebody else should probably review this too. 

I would suggest that when you address the comments you rebase and squash the commits so that it doesn't have any merge (or otherwise superflous) commits.

File: example/fcn-xs/README.md (hunk 1 above)
Comment (antinucleon) [inline example/fcn-xs/README.md]: We can commit image to ```dmlc/web-data``` to reduce repo size

**Comment by piiswrong** [comment]
Good work! Think this is a really good example. It illustrates how to write an DataIter for custom data and how to write custom solver for custom model.

We have got a lot of questions for how to do this. It would be great if you can summarize your work into a tutorial.
Also, given that you have to write a new solver, what do you think about the _train_multi_device refactoring here https://github.com/dmlc/mxnet/pull/909? Maybe you can refactor your solver to use ExecutorManager and improve the interface in the process?

**Comment by tornadomeet** [comment]
hi, all, this examples changed some code of mashadow and mxnet specificly for seg , so there is some style we should debat,. btw,  the code is some dirty, i'll rearrange the sample in this weekend~ so please wait for me some days, thanks.

**Comment by tqchen** [comment]
close this PR till @tornadomeet 's update

**Pull request closed without merging**